plugin/pkg/admission/namespace/lifecycle/admission_test.go

```diff
@@ -0,0 +1,183 @@
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/internalclientset"
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/internalclientset/fake"
+	informers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers_generated/internalversion"
+
+	scadmission "github.com/kubernetes-incubator/service-catalog/pkg/apiserver/admission"
+)
+
+// newHandlerForTest returns a configured handler for testing.
+func newHandlerForTest(internalClient internalclientset.Interface, kubeClient kubeclientset.Interface) (admission.Interface, informers.SharedInformerFactory, kubeinformers.SharedInformerFactory, error) {
+	f := informers.NewSharedInformerFactory(internalClient, 5*time.Minute)
+	kf := kubeinformers.NewSharedInformerFactory(kubeClient, 5*time.Minute)
+	handler, err := NewLifecycle()
+	if err != nil {
+		return nil, f, kf, err
+	}
+	pluginInitializer := scadmission.NewPluginInitializer(internalClient, f, kubeClient, kf)
+	pluginInitializer.Initialize(handler)
+	err = admission.Validate(handler)
+	return handler, f, kf, err
+}
+
+// newMockKubeClientForTest creates a mock client that returns a client
+// configured for the specified list of namespaces with the specified phase.
+func newMockKubeClientForTest(namespaces map[string]v1.NamespacePhase) *kubefake.Clientset {
+	mockClient := &kubefake.Clientset{}
+	mockClient.AddReactor("list", "namespaces", func(action core.Action) (bool, runtime.Object, error) {
+		namespaceList := &v1.NamespaceList{
+			ListMeta: metav1.ListMeta{
+				ResourceVersion: fmt.Sprintf("%d", len(namespaces)),
+			},
+		}
+		index := 0
+		for name, phase := range namespaces {
+			namespaceList.Items = append(namespaceList.Items, v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            name,
+					ResourceVersion: fmt.Sprintf("%d", index),
+				},
+				Status: v1.NamespaceStatus{
+					Phase: phase,
+				},
+			})
+			index++
+		}
+		return true, namespaceList, nil
+	})
+	return mockClient
+}
+
+// newMockClientForTest creates a mock client.
+func newMockClientForTest() *fake.Clientset {
+	mockClient := &fake.Clientset{}
+	return mockClient
+}
+
+// newBroker returns a new broker for testing.
+func newBroker() servicecatalog.Broker {
+	return servicecatalog.Broker{
+		ObjectMeta: metav1.ObjectMeta{Name: "broker"},
+	}
+}
+
+// newInstance returns a new instance for the specified namespace.
+func newInstance(namespace string) servicecatalog.Instance {
+	return servicecatalog.Instance{
+		ObjectMeta: metav1.ObjectMeta{Name: "instance", Namespace: namespace},
+	}
+}
+
+// TestAdmissionNamespaceDoesNotExist verifies instance is not admitted if namespace does not exist.
+func TestAdmissionNamespaceDoesNotExist(t *testing.T) {
+	namespace := "test"
+	mockClient := newMockClientForTest()
+	mockKubeClient := newMockKubeClientForTest(map[string]v1.NamespacePhase{})
```

Comment:
general commentary. statement of a fact, that hopefully someone would correct me if I'm wrong.